library/std/src/sys/hermit/condvar.rs

```diff
@@ -1,80 +1,108 @@
 use crate::ffi::c_void;
+use crate::mem::MaybeUninit;
 use crate::ptr;
-use crate::sync::atomic::{AtomicUsize, Ordering::SeqCst};
+use crate::sync::atomic::{
+    AtomicPtr, AtomicUsize,
+    Ordering::{Acquire, Release, SeqCst},
+};
 use crate::sys::hermit::abi;
 use crate::sys::locks::Mutex;
-use crate::sys_common::lazy_box::{LazyBox, LazyInit};
 use crate::time::Duration;
 
 // The implementation is inspired by Andrew D. Birrell's paper
 // "Implementing Condition Variables with Semaphores"
 
 pub struct Condvar {
     counter: AtomicUsize,
-    sem1: *const c_void,
-    sem2: *const c_void,
+    sem1: AtomicPtr<c_void>,
+    sem2: AtomicPtr<c_void>,
 }
 
-pub(crate) type MovableCondvar = LazyBox<Condvar>;
+pub(crate) type MovableCondvar = Condvar;
 
-impl LazyInit for Condvar {
-    fn init() -> Box<Self> {
-        Box::new(Self::new())
+#[cold]
+fn init_semaphore(sem: &AtomicPtr<c_void>) -> *mut c_void {
+    let new = unsafe {
+        let mut new = MaybeUninit::uninit();
```

Comment:
Using MaybeUninit for this is overkill (it's just a pointer) and would result in strange behavior if the `sem_init` call fails. Given that you ignore it's error, that seems bad.